### PR TITLE
added extended model

### PR DESCRIPTION
**Pull request by Juan-EBI**

### CODE REVIEW

File: commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt [hunk 1]
```diff
@@ -39,12 +78,13 @@ var Submission.rootPath: String?
...
     set(value) {
         value?.let { this[SubFields.TITLE] = it }
     }
 
+fun Submission.addAccessTag(accessTag: String) = accessTags.add(accessTag)
 fun Submission.allFiles() = section.allFiles() + section.allSections().flatMap { it.allFiles() }
-fun Submission.getSectionByType(name: String) = section.allSections().first { it.type == name }
-fun Submission.fileListSections() = (section.allSections() + section).filterNot { it.fileListName.isNullOrBlank() }
+fun Submission.getSectionByType(name: String): Section = section.allSections().first { it.type == name }
+fun Submission.fileListSections() = (section.allSections() + section).filterNot { it.fileListName == null }
```
Comment (jhoanmanuelms) [inline commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt]: For this case, if we had an empty file list it shouldn't be considered a `fileListSection`, that's why it was `isNullOrBlank` before

File: commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/from/ToExtAttribute.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import ebi.ac.uk.extended.model.ExtAttribute
+import ebi.ac.uk.extended.model.ExtAttributeDetail
+import ebi.ac.uk.model.Attribute
+import ebi.ac.uk.model.AttributeDetail
+
+fun Attribute.toExtAttribute() = ExtAttribute(name, value, reference, toDetails(nameAttrs), toDetails(valueAttrs))
+
+private fun toDetails(details: List<AttributeDetail>) = details.map { ExtAttributeDetail(it.name, it.value) }
+
+internal const val TO_EXT_ATTRIBUTE_EXTENSIONS = "ebi.ac.uk.extended.mapping.serialization.from.ToExtAttributeKt"
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/from/ToExtAttribute.kt]: This is just cosmetic bu, It looks a bit weird to me to have a constant at the bottom of the file 😅

File: commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/to/ToSubmission.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.model.Submission
+import ebi.ac.uk.model.extensions.creationTime
+import ebi.ac.uk.model.extensions.modificationTime
+import ebi.ac.uk.model.extensions.releaseTime
+import ebi.ac.uk.model.extensions.secretKey
+
+/**
+ * Return a simple submission which do not contains submission secret information.
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/to/ToSubmission.kt]: does not contain*

File: commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/to/ToSubmission.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        attributes = attributes.map { it.toAttribute() }.toMutableList(),
+        tags = tags.toMutableList()
+    )
+}
+
+/**
+ * Return a extended submission which contains submission secret information and internal information.
+ */
+fun ExtSubmission.toExtSubmission(): Submission {
+    val submission = Submission(
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/to/ToSubmission.kt]: This one is called `toExtSubmission` but returns a regular `Submission`. Is this intended?

File: commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/to/ToSubmission.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        accNo = accNo,
+        section = section.toSection(),
+        attributes = attributes.map { it.toAttribute() }.toMutableList(),
+        tags = tags.toMutableList(),
+        accessTags = accessTags.toMutableList()
+    )
+
+    submission.secretKey = secretKey
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/to/ToSubmission.kt]: No `apply`? 😮

File: commons/commons-model-extended/src/test/kotlin/ebi/ac/uk/extended/mapping/serialization/from/ToExtFileTest.kt [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    @MockK val filesSource: FilesSource,
+    @MockK val attribute: Attribute,
+    @MockK val extAttribute: ExtAttribute,
+    @MockK val systemFile: SystemFile
+) {
+    private val file = File("fileName", 55L, listOf(attribute))
+
+    @Test
+    fun toExtFile() {
+        mockkStatic(TO_EXT_ATTRIBUTE_EXTENSIONS) {
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended/src/test/kotlin/ebi/ac/uk/extended/mapping/serialization/from/ToExtFileTest.kt]: Fancy :bowtie:

File: commons/commons-model-extended/src/test/kotlin/ebi/ac/uk/extended/mapping/serialization/from/ToExtFileTest.kt [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended/src/test/kotlin/ebi/ac/uk/extended/mapping/serialization/from/ToExtFileTest.kt]: Extra line

File: commons/commons-model-extended/src/test/kotlin/ebi/ac/uk/extended/mapping/serialization/from/ToExtLinkTest.kt [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import io.mockk.impl.annotations.MockK
+import io.mockk.junit5.MockKExtension
+import io.mockk.mockkStatic
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(MockKExtension::class)
+internal class ToExtLinkTest(
+    @MockK val attribute: Attribute,
```
Comment (jhoanmanuelms) [inline commons/commons-model-extended/src/test/kotlin/ebi/ac/uk/extended/mapping/serialization/from/ToExtLinkTest.kt]: This should fit in one line shouldn't it?

File: commons/commons-util/src/main/kotlin/ebi/ac/uk/util/collections/ListExtensions.kt [hunk 9]
```diff
@@ -4,6 +4,14 @@ import java.util.LinkedList
 
 private const val NO_FOUND = -1
 
+inline fun <T> List<T>.indexOf(predicate: (T) -> Boolean): Int? {
```
Comment (jhoanmanuelms) [inline commons/commons-util/src/main/kotlin/ebi/ac/uk/util/collections/ListExtensions.kt]: We should add docs for this one

File: submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/assertions/SubmissionAssertHelper.kt [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+import ebi.ac.uk.model.Section
+import ebi.ac.uk.model.SectionsTable
+import ebi.ac.uk.model.attributeDetails
+import ebi.ac.uk.util.collections.second
+import ebi.ac.uk.util.collections.third
+import org.assertj.core.api.Assertions.assertThat
+import java.nio.file.Paths
+
+internal class SubmissionAssertHelper(private val basePath: String) {
+
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/assertions/SubmissionAssertHelper.kt]: Blank

File: submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/assertions/SubmissionAssertHelper.kt [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+import ebi.ac.uk.model.Link
+import ebi.ac.uk.model.Section
+import ebi.ac.uk.model.SectionsTable
+import ebi.ac.uk.model.attributeDetails
+import ebi.ac.uk.util.collections.second
+import ebi.ac.uk.util.collections.third
+import org.assertj.core.api.Assertions.assertThat
+import java.nio.file.Paths
+
+internal class SubmissionAssertHelper(private val basePath: String) {
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/assertions/SubmissionAssertHelper.kt]: Great! 🤘

File: commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt (hunk 1 above)
Comment (Juan-EBI) [reply commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/extensions/SubExt.kt]: if the `fileListName` is present but empty this is an error and should be trigger or system should break. it should not be possible to have an empty `fileListName`

File: commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/to/ToSubmission.kt (hunk 4 above)
Comment (Juan-EBI) [reply commons/commons-model-extended/src/main/kotlin/ebi/ac/uk/extended/mapping/serialization/to/ToSubmission.kt]: this function has two version one that includes public info (intended for file serialization and user api) and another that contains secret information (intended to be used by frontend) .

that is the reason that name do not match the return type. Any suggestion for naming?

**Pull request merged**